anon/elgam.go

```diff
@@ -0,0 +1,82 @@
+package anon
+
+import (
+	"bytes"
+	"errors"
+	"crypto/cipher"
+	"github.com/dedis/crypto/abstract"
+)
+
+type basicSig struct {
+	C0 abstract.Secret
+	S  abstract.Secret
+}
+
+func signH1preElGam(suite abstract.Suite, message []byte) []byte {
+	H := suite.Hash()
+	H.Write(message)			
+	return H.Sum(nil)
+}
+
+func signH1ElGam(suite abstract.Suite, H1pb []byte, PG abstract.Point) abstract.Secret {
+	H1 := suite.Hash()
+	H1.Write(H1pb)
+	H1.Write(PG.Encode())
+
+	b := H1.Sum(nil)
+	s := suite.Stream(b[:suite.KeyLen()])
+	return suite.Secret().Pick(s)
+}
+
+
+func SignElGam(suite abstract.Suite, random cipher.Stream, message []byte,
+	publicKey abstract.Point , privateKey abstract.Secret) []byte {
+
+	// generate signature
+	H := signH1preElGam(suite, message)
+
+	u := suite.Secret().Pick(random)
+	UB := suite.Point().Mul(nil,u)
+	
```

Comment:
Where do these variable names come from?  Unless they're intended to match up with some very well-known, definitive formulation of ElGamal, I would prefer if variable naming followed the Camenisch-Stadler terminology: e.g., v for the random private value, T for the public point commitment to that value, just c rather than c0 for the challenge, and r rather than s for the response.  

On the other hand, if you did find some very definitive formulation, it might indeed be better to be consistent with that.